### ISSUE TITLE
Need better events for face color

### ISSUE THREAD

**Issue by VasudhaJha** [napari/napari#4573]
## 🚀 Feature

Can we get better events for face color that work for both points and shapes layers?

## Motivation

We are working on a plugin that displays different layers (points, shapes, label) as a table. In this table, we are also showing a point's face color and whenever it is changed in the napari viewer, we are modifying it in the table as well. However, we are not getting the required events to do so.

The `face_color` event (`<layer>.events.face_color.connect(...)`) does not get triggered when face color is changed. The `current_face_color` event (`<layer>.events.current_face_color.connect(...)`) gets triggered, but only for shapes layer. In order to get the face color events for points layer, our workaround is to use the `current_color` event for the layer's internal `_face` (`<layer>._face.events.current_color.connect(...)`).

## Pitch

I would like to have one event that can be subscribed to for getting face color changes that works for all layers. Can the existing `face_color` event be fixed?

## Alternatives

- For shapes layer, we're connecting to the `current_face_color` event (`<layer>.events.current_face_color.connect(...)`).
- For points layer, we're connecting to the `current_color` event for the layer's internal `_face` (`<layer>._face.events.current_color.connect(...)`).

**Comment by cudmore** [comment]
For what it is worth, I was just poking around in the vispy code and see it is relying on a number of face color events.

See file: [napari/_vispy/layers/points.py](https://github.com/napari/napari/blob/main/napari/_vispy/layers/points.py)
```
self.layer.events.face_color.connect(self._on_data_change)
self.layer._face.events.colors.connect(self._on_data_change)
self.layer._face.events.color_properties.connect(self._on_data_change)
```

**Comment by andy-sweet** [comment]
Thanks for filing the issue! This might be solved with some upcoming work, though I'm not sure yet.

Are you manually assigning colors to points? E.g. by selecting some and changing the current color either with the API or the GUI using the color picker?

Or are you assigning colors to points from properties? As in this tutorial: https://napari.org/howtos/layers/points.html#setting-point-edge-and-face-color-with-properties

**Comment by VasudhaJha** [comment]
> Are you manually assigning colors to points? E.g. by selecting some and changing the current color either with the API or the GUI using the color picker?

Yes, I'm selecting one or more points manually and changing their color from the GUI using the color picker and also from the API.
